### PR TITLE
=str Update FlowMergeAllSpec because the merge order is undefined.

### PR DESCRIPTION
**Pull request by He-Pin**
References https://github.com/akka/akka/issues/31546

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowMergeAllSpec.scala [hunk 1]
```diff
@@ -27,10 +27,11 @@ class FlowMergeAllSpec extends StreamSpec("""
 
     "merge all elements of the first completed source to its downstream " in {
       val source1 = Source(1 to 2)
-      val source2 = Source(3 to 6)
-      val source3 = Source(7 to 10)
+      val source2 = Source.repeat(3)
+      val source3 = Source.repeat(4)
```
Comment (He-Pin) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowMergeAllSpec.scala:31]: Becase the merge order is undefined.

**Review by johanandren — APPROVED** [review]
LGTM (but a test with source probes might have been better to start with)

**Comment by He-Pin** [comment]
@patriknw @johanandren I think this one can be merged.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**